governance/program/src/state/governance.rs

```diff
@@ -377,20 +428,181 @@ pub fn assert_valid_create_governance_args(
 pub fn assert_is_valid_governance_config(
     governance_config: &GovernanceConfig,
 ) -> Result<(), ProgramError> {
-    match governance_config.community_vote_threshold {
+    assert_is_valid_vote_threshold(&governance_config.community_vote_threshold)?;
+    assert_is_valid_vote_threshold(&governance_config.council_vote_threshold)?;
+
+    if governance_config.community_vote_threshold == VoteThreshold::Disabled
+        && governance_config.council_vote_threshold == VoteThreshold::Disabled
+    {
+        return Err(GovernanceError::AllVoteThresholdsDisabled.into());
+    }
+
+    if governance_config.reserved != [0, 0] {
+        return Err(GovernanceError::ReservedBufferMustBeEmpty.into());
+    }
+
+    Ok(())
+}
+
+/// Asserts the provided vote_threshold is valid
+pub fn assert_is_valid_vote_threshold(vote_threshold: &VoteThreshold) -> Result<(), ProgramError> {
+    match *vote_threshold {
         VoteThreshold::YesVotePercentage(yes_vote_threshold_percentage) => {
             if !(1..=100).contains(&yes_vote_threshold_percentage) {
```

Comment:
It turns out really well that `0` is invalid!

Reply:
yeah :)